whois-api/src/main/java/net/ripe/db/whois/api/rdap/RedactionObjectMapper.java

```diff
@@ -33,28 +40,43 @@ public static void mapRedactions(final RdapObject rdapObject) {
         addEntitiesRedaction(rdapObject, rdapObject.getEntitySearchResults(), "$");
     }
 
-    private static void addEntityRedaction(final RdapObject rdapObject) {
-        if(rdapObject instanceof Entity) {
-            rdapObject.getRedacted().addAll(getPersonalRedaction( ((Entity) rdapObject).getvCardRedactedAttr(), "$"));
-        }
-    }
-
     private static void addEntitiesRedaction(final RdapObject rdapObject, final List<Entity> entities, final String prefix) {
         entities.forEach( entity -> {
-            rdapObject.getRedacted().addAll(getPersonalRedaction(entity.getvCardRedactedAttr(), String.format("%s.entities[?(@.handle=='%s')]", prefix, entity.getHandle())));
+            rdapObject.getRedacted().addAll(getRedactions(entity.getRedactedRpslAttrs(), String.format("%s.entities[?(@.handle=='%s')]", prefix, entity.getHandle())));
         });
     }
 
-    private static Set<Redaction> getPersonalRedaction(final List<AttributeType> attributeTypes, final String prefix){
+    private static Set<Redaction> getRedactions(final List<RpslAttribute> rpslAttributes, final String prefix){
         final Set<Redaction> redactions = Sets.newHashSet();
-        attributeTypes.forEach(attributeType -> {
-            switch(attributeType){
-                case NOTIFY -> redactions.add(new Redaction("Updates notification e-mail information", String.format("%s.vcardArray[1][?(@[0]=='%s')]", prefix, attributeType.getName()),
-                        "Personal data"));
-                case E_MAIL -> redactions.add(new Redaction("Personal e-mail information", String.format("%s.vcardArray[1][?(@[0]=='%s')]", prefix, attributeType.getName()),
-                        "Personal data"));
+
+        final Map<AttributeType, List<CIString>> attributeTypeByValues =  rpslAttributes.stream()
+                .collect(Collectors.groupingBy(RpslAttribute::getType, Collectors.mapping(RpslAttribute::getCleanValue, Collectors.toList())));
+
+        attributeTypeByValues.entrySet().forEach(entry -> {
+            final String attributeName = entry.getKey().getName();
+            final String values = entry.getValue().stream().collect(Collectors.joining(", "));
```

Comment:
after previous change you no longer need to get the values from the entry.
so you can directly:
final String values = String.join(", ", value)